internal/ls/signaturehelp.go

```diff
@@ -0,0 +1,1123 @@
+		}
+	} else {
+		activeParameter = ptrTo(lsproto.ToNullable(uint32(*argumentInfo.argumentIndex)))
+	}
+	return &lsproto.SignatureHelp{
+		Signatures:      signatureInformation,
+		ActiveSignature: ptrTo(uint32(0)),
+		ActiveParameter: activeParameter,
+	}
+}
+
+func getTypeHelpItem(symbol *ast.Symbol, typeParameter []*checker.Type, enclosingDeclaration *ast.Node, sourceFile *ast.SourceFile, c *checker.Checker) signatureInformation {
+	printer := printer.NewPrinter(printer.PrinterOptions{NewLine: core.NewLineKindLF}, printer.PrintHandlers{}, nil)
+
+	var symbolString strings.Builder
+	symbolString.WriteString(c.SymbolToString(symbol))
+
+	var parameters []signatureHelpParameter = []signatureHelpParameter{}
+	for _, typeParam := range typeParameter {
+		parameters = append(parameters, createSignatureHelpParameterForTypeParameter(typeParam, sourceFile, enclosingDeclaration, c, printer))
+	}
+
+	// Creating display label
+	typeParameterDisplayParts := []string{}
+	for _, typeParameter := range parameters {
+		typeParameterDisplayParts = append(typeParameterDisplayParts, *typeParameter.parameterInfo.Label.String)
+	}
+	typeParameterParts := []string{}
+	if len(typeParameterDisplayParts) != 0 {
+		typeParameterParts = []string{scanner.TokenToString(ast.KindLessThanToken), strings.Join(typeParameterDisplayParts, ", "), scanner.TokenToString(ast.KindGreaterThanToken)}
+	}
+	displayParts := []string{}
+	displayParts = append(displayParts, symbolString.String())
+	displayParts = append(displayParts, typeParameterParts...)
+
+	return signatureInformation{
+		Label:         strings.Join(displayParts, ""),
+		Documentation: nil,
+		Parameters:    &parameters,
+		IsVariadic:    false,
+	}
+}
+
+func createSignatureHelpItems(candidates *[]*checker.Signature, resolvedSignature *checker.Signature, argumentInfo *argumentListInfo, sourceFile *ast.SourceFile, c *checker.Checker, useFullPrefix bool, clientOptions *lsproto.SignatureHelpClientCapabilities) *lsproto.SignatureHelp {
+	enclosingDeclaration := getEnclosingDeclarationFromInvocation(argumentInfo.invocation)
+	if enclosingDeclaration == nil {
+		return nil
+	}
+	var callTargetSymbol *ast.Symbol
+	if argumentInfo.invocation.contextualInvocation != nil {
+		callTargetSymbol = argumentInfo.invocation.contextualInvocation.symbol
+	} else {
+		callTargetSymbol = c.GetSymbolAtLocation(getExpressionFromInvocation(argumentInfo))
+		if callTargetSymbol == nil && useFullPrefix && resolvedSignature.Declaration() != nil {
+			callTargetSymbol = resolvedSignature.Declaration().Symbol()
+		}
+	}
+
+	callTargetDisplayParts := []string{}
+	if callTargetSymbol != nil {
+		callTargetDisplayParts = append(callTargetDisplayParts, c.SymbolToString(callTargetSymbol))
+	}
+	var items [][]signatureInformation
+	for _, candidateSignature := range *candidates {
+		items = append(items, getSignatureHelpItem(candidateSignature, argumentInfo.isTypeParameterList, callTargetDisplayParts, enclosingDeclaration, sourceFile, c))
```

Comment:
This is the only call to `getSignatureHelpItem`, and `callTargetDisplayParts` is always length 0 or 1. It could just be a string. We know we're not going to have display parts anymore, so it doesn't seem like there's a need to keep the slices around. A lot of things can probably just become strings.